app/components/panel.rb

```diff
@@ -0,0 +1,226 @@
+  prop :open, _Boolean, default: false
+  prop :panel_bodies, _Nilable(Array), default: nil
+  prop :formatted_content, _Boolean, default: false
+  prop :attributes, Hash, default: -> { {} }
+
+  def view_template(&block)
+    content = block ? capture(&block).to_s : ""
+
+    # If inner_id is provided and there's a heading, put the ID on the outer
+    # div so tests can scope to the entire panel including heading links
+    panel_id = @inner_id if @heading.present?
+
+    div(
+      class: class_names("panel panel-default", @panel_class),
+      id: panel_id,
+      **@attributes
+    ) do
+      render_heading if @heading
+      render_thumbnail if @thumbnail
+      render_body_or_bodies(content)
+      render_footer if @footer
+    end
+  end
+
+  private
+
+  def render_heading
+    div(class: "panel-heading") do
+      if @collapse.present?
+        render_collapsible_heading
+      else
+        render_static_heading
+      end
+    end
+  end
+
+  def render_static_heading
+    h4(class: "panel-title") do
+      # heading may contain HTML (e.g., submit buttons, formatted text) that
+      # needs to be rendered as HTML, not escaped as text
+      # rubocop:disable Rails/OutputSafety
+      raw(@heading.html_safe)
+      if @heading_links.present?
+        # heading_links contains HTML from link_to helper
+        # that needs to be rendered as HTML, not escaped as text
+        span(class: "float-right") { raw(@heading_links.html_safe) }
+      end
+      # rubocop:enable Rails/OutputSafety
+    end
+  end
+
+  def render_collapsible_heading
+    collapsed_class = @open ? "" : "collapsed"
+
+    h4(class: "panel-title") do
+      link_to(
+        "##{@collapse}",
+        class: class_names("panel-collapse-trigger", collapsed_class),
+        role: "button",
+        data: { toggle: "collapse" },
+        aria: { expanded: @open, controls: @collapse }
+      ) do
+        # heading may contain HTML (e.g., submit buttons, formatted text) that
+        # needs to be rendered as HTML, not escaped as text
+        # rubocop:disable Rails/OutputSafety
+        raw(@heading.html_safe)
+        # rubocop:enable Rails/OutputSafety
+        span(class: "float-right") { render_collapse_icons }
+      end
+    end
+  end
+
+  def render_collapse_icons
+    if @collapse_message.present?
+      span(class: "font-weight-normal mr-2") do
+        plain(@collapse_message)
+      end
+    end
+
+    chevron_down = link_icon(:chevron_down, title: :OPEN.l,
+                                            class: "active-icon")
+    chevron_up = link_icon(:chevron_up, title: :CLOSE.l)
+
+    # link_icon returns HTML strings that need to be rendered as HTML,
+    # not escaped as text. Nil check needed as link_icon may return nil
+    # in test environments where icon assets aren't available
+    # rubocop:disable Rails/OutputSafety
+    raw(chevron_down.html_safe) if chevron_down
+    raw(chevron_up.html_safe) if chevron_up
```

Comment:
Dug into these cases and apparently none of this is needed any more because of Phlex.  Replacing lines 143 to 153 with just:
```
    link_icon(:chevron_down, title: :OPEN.l,
              class: "active-icon")
    link_icon(:chevron_up, title: :CLOSE.l)
```
appears to do the job.

Reply:
This change is also on njw-phlex-discoveries